JasperReports/src/org/compiere/report/ReportStarter.java

```diff
@@ -519,26 +520,38 @@ else if(
 
             if (Record_ID > 0) {
             	params.put("RECORD_ID", new Integer(Record_ID));
-            	params.put("Record_ID", new Integer(Record_ID));
+            	if(!Optional.ofNullable(params.get("Record_ID")).isPresent()) {

```

Comment:
Consider .OrElse() to remove the "if not" statement as in

    Optional.ofNullable(params.get("Record_ID")).orElse(
        params.put("Record_ID", new Integer(Record_ID)));

You could also extract these repeated patterns to a method as so

    protected void verifyParameterExistsOrAddDefault(
            HashMap<String, Object> params, final String parameterName,
            int defaultValue) {

        Optional.ofNullable(params.get(parameterName)).orElse(
            params.put(parameterName, new Integer(defaultValue)));

    }

Then it would look like

            if (Record_ID > 0) {
            	params.put("RECORD_ID", new Integer(Record_ID));
            	verifyParameterExistsOrAddDefault(params, "Record_ID", Record_ID);
            }